chopper.py

```diff
@@ -326,92 +346,81 @@ def combine_files(in_files: list[Path], config: Config) -> Path:
     if len(in_files) == 1:
         return in_files[0]
 
-    combined_fp = config.output_dir / "combined"
-    with open(combined_fp, "w", encoding=config.encoding) as fout:
+    combined_fp = config.output_directory / "combined"
+    with combined_fp.open("w", encoding=config.encoding, newline="") as fout:
         for i, f in enumerate(in_files):
-            with open(f, "r") as fin:
+            with f.open("r", encoding=config.encoding, newline="") as fin:
                 if i > 0:  # Skip the header row, except for the first file.
-                    next(fin)
+                    fin.readline()
                 for line in fin:
                     fout.write(line)
-
     return combined_fp
 
 
 def main() -> None:
-    args = parse_args()
-    input_path: Path = args["input_path"]
-    output_dir: Path = args["output_directory"]
-    extension: str = args["extension"]
-    columns: str = args["columns"]
-    rows: int = args["rows"]
-    encoding: str = args["encoding"]
-    prefix: str = args["prefix"]
-    shuffles: int = args["shuffles"]
-    delimiter: str = args["delimiter"]
-    equal: int = args["equal"]
-
-    config = Config(encoding, delimiter, output_dir)
+    config = parse_args()
 
     files = []
     output_ext = ".csv"  # Default. Overridden below based on actual files.
 
-    if input_path.is_dir():
-        # "*" is the default extension for wildcarding.
-        in_files = list(input_path.glob(f"**/*.{extension}"))
-
-        # Use the extension of the first matching file. Useful when extension is not specified.
-        output_ext = in_files[0].suffix
-
-        files = [combine_files(in_files, config)]
-        config.is_original = False
+    if len(config.input_paths) == 1 and config.input_paths[0].is_file():
+        output_ext = config.input_paths[0].suffix
+        files.append(config.input_paths[0])
     else:
-        # If input is a file, use that file's extension for outputs.
-        output_ext = input_path.suffix
-        files.append(input_path)
+        combine_me = []
+        for file in config.input_paths:
+            if file.is_dir():
+                # "*" is the default extension for wildcarding.
+                in_files = list(file.glob(f"**/*.{config.extension}"))
+
+                # Use the extension of the first matching file. Useful when extension is not specified.
+                output_ext = in_files[0].suffix
+                combine_me.extend(in_files)
+            else:
+                # If input is a file, use that file's extension for outputs.
+                output_ext = file.suffix
+                combine_me.append(file)
+        if len(combine_me) == 1:
+            files.append(combine_me[0])
+        else:
+            files.append(combine_files(combine_me, config))
```

Comment:
Reasonably big changes here to support multiple file/directory inputs and to clean up the logic of when to skip the combine step. This will soon change a bit again because one of my TODO items is to bypass the combine step altogether when `--shuffles` is not used.